spec/lang/representation.md

```diff
@@ -229,16 +229,54 @@ impl Type {
 
 ### Enums
 
-TODO: implement Enum decoding & encoding.
+Enum encoding and decoding.
+Note that the discriminant may not be written into bytes that contain encoded data.
+This is to ensure that pointers to the data always contain valid values.
 
 ```rust
+/// Computes the discriminant from the value's bytes by the decision tree in the `Discriminator` by
+/// recursing over the branches defined by `Discriminator::Unknown`.
+/// FIXME: we have multiple quite different fail sources, it would be nice to return more error information.
+fn compute_discriminant<M: Memory>(bytes: List<AbstractByte<M::Provenance>>, discriminator: Discriminator) -> Option<Int> {
```

Comment:
"compute" is not entirely accurate, we're not doing much computation here after all. I think this should be called "decode_discriminant"; that also makes it clear which operation this is a helper for.